src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -551,19 +550,11 @@ public void SetNonBlocking()
             }
         }
 
-        public void CheckForPriorConnectFailure()
-        {
-            if (_connectFailed)
-            {
-                throw new PlatformNotSupportedException(SR.net_sockets_connect_multiconnect_notsupported);
-            }
-        }
-
         public void RegisterConnectResult(SocketError error)
         {
             if (error != SocketError.Success && error != SocketError.WouldBlock)
             {
-                _connectFailed = true;
+                _socket.LastConnectFailed = true;
```

Comment:
Since this function modifies state on the SafeCloseSocket, would it make more sense for it to live on that class?